src/ModelObjects/component.rs

```diff
@@ -45,6 +45,22 @@ impl DeclarationProvider for Component {
 
 #[allow(dead_code)]
 impl Component {
+    pub fn invalid() -> Self {
+        Component {
+            name: "Invalid component".to_string(),
+            declarations: Declarations::empty(),
+            locations: vec![],
+            edges: vec![],
+            input_edges: Some(vec![]),
+            output_edges: Some(vec![]),
+        }
+    }
```

Comment:
Unused currently, and should probably be removed?

Reply:
Woopsie I left in some of the earlier code, I'm removing it